tests/README.md

```diff
@@ -0,0 +1,1510 @@
+### Test suites
+Currently there are three different test suites:
+- ILIAS test suite which runs all ILIAS unbound unit tests
+- ILIAS performance tests
+- ILIAS static code analysis (done with Dicto.php)
+
+The ILIAS unit test suite runs agains the following PHP versions:
+- 5.6
+- 7.0
+- 7.1
+
+<a name="benefits"></a>
+### Benefits
+There are several benefits, first the tests are running in clean defined environment carefully monitored and maintained by the community, second all developer are informed about the actual condition of ILIAS which allows to react fast to emerging problems.
+
+<a name="current-problems"></a>
+### Current problems
+The CI server uses different configuration which could lead to confusing results for the developer which checked-in the code.
+
+<a name="external-phpunit-version"></a>
+#### External PHPUnit version
+The CI server uses a external php unit version which is not updated by composer
+which could lead to different results.
+
+The current command of the php 7 worker looks like this:
+
+```
+/usr/bin/php7.0 /usr/local/bin/phpunit-5.7.20.phar /
+	--log-junit %system.teamcity.build.tempDir%/phpunit-log-7.0.x.xml /
+	-c  %system.teamcity.build.workingDir%/Services/PHPUnit/config/PhpUnitConfig.xml
+```
+
+But should look like this to leverage the composer version of php unit:
+
+```
+/usr/bin/php7.0 ./libs/composer/vendor/bin/phpunit /
+	--log-junit %system.teamcity.build.tempDir%/phpunit-log-7.0.x.xml /
+	-c  %system.teamcity.build.workingDir%/Services/PHPUnit/config/PhpUnitConfig.xml
+```
+
+<a name="risky-prebuild-step"></a>
+#### Risky prebuild step
+The CI server runs commands before the test suite to ensure a known state of the
+environment.
+
+The prebuild step looks like this:
+
+```
+cd %system.teamcity.build.workingDir%/libs/composer && rm -rf vendor/geshi/ && /usr/bin/composer update --prefer-dist
+```
+
+Composer has a special file called composer.lock which ensures that the dependencies
+are the same on each host. This lock file is red at install time to download all dependencies. However the composer update command ignores and overwrites the lock file to update the dependencies to the newest version. That could lead to unpredictable
+results because the dependencies may be newer than the ones the developer used
+at check-in time.
+
+The command should look like this:
+
+```
+cd %system.teamcity.build.workingDir%/libs/composer && rm -rf vendor/geshi/ && /usr/bin/composer install
+```
+
+<a name="external-documentation"></a>
+## External Documentation
+[PHP Unit Documentation](https://phpunit.de/manual/5.7/en/index.html)
+
+[Mockery Documentation](http://docs.mockery.io/en/latest/)
+
+[ILIAS CI Server](https://ci.ilias.de/)
+
+[xDebug Documentation](https://xdebug.org/docs/)
+
+[Dicto.php](https://github.com/lechimp-p/dicto.php)
+
+<a name="old-documentation"></a>
+## Old Documentation
+[Old ILIAS Unit Test Guide](https://www.ilias.de/docu/goto_docu_pg_29759_42.html)
+
+<a name="faq"></a>
+## FAQ
```

Comment:
I would suggest to put that more to the front. This seems to serve as another point to dive into the rest of the docu.

Reply:
I am not really sure where I should put the FAQ, any suggestions ?

Reply:
I would suggest to put it as he first chapter of this guide.